perf/tuple.cc

```diff
@@ -72,6 +105,32 @@ access_tuple_data(benchmark::State& state)
 }
 BENCHMARK(access_tuple_data);
 
+static void
+access_tiny_tuple_data(benchmark::State& state)
```

Comment:
I think this approach does not represent real situation. With what will you compare the test results? With performance of big tuples? They could be different just because tuple sizes are different, not because of tiny tuple format.
I think that the tests should measure performance of different cases - small tuples, big tuples, mixed sizes. And then you can run the test before your implementation of tiny tuples and after and compare.